shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/util/SQLUtil.java

```diff
@@ -280,12 +279,12 @@ public static String trimComment(final String sql) {
     }
     
     /**
-     * Get show like pattern.
+     * Replace all SQL pattern to java regex.
      * 
-     * @param showLike show like segment
-     * @return pattern
+     * @param pattern SQL pattern
+     * @return java regex
      */
-    public static String getShowLikePattern(final ShowLikeSegment showLike) {
-        return showLike.getPattern().replaceAll("_", ".{1}").replaceAll("%", ".*");
+    public static String handleSQLPattern(final String pattern) {
+        return pattern.contains("_") || pattern.contains("%") ? pattern.replaceAll("_", ".").replaceAll("%", ".*") : pattern;
```

Comment:
Please consider if the original varchar contains `%` or `_`

Reply:
> contains

@terrymanu Thank you for your remind, i will check official doc.